docs/src/manual/troubleshooting.md

```diff
@@ -1,5 +1,49 @@
 # Troubleshooting
 
+Jupyter notebooks run code by connecting to a [kernel](https://docs.jupyter.org/en/stable/projects/kernels.html). That's exactly what the `IJulia` package provides. Jupyter knows about available kernels through [kernel specs](https://jupyter-client.readthedocs.io/en/stable/kernels.html#kernel-specs), which are `kernel.json` files inside [Jupyter's data paths](https://docs.jupyter.org/en/stable/use/jupyter-directories.html#data-files). You can list the available kernel specs on the command line with
+
+```sh
+jupyter kernelspec list
+```
+
+Or, explore the data directory relevant to your system, e.g., `~/.local/share/jupyter/kernels`.
+
+Make sure that you can find a Julia kernel. If you can't, run [`IJulia.installkernel`](@ref), e.g., as `import IJulia; IJulia.installkernel("Julia", "--project=@.")` in the Julia REPL.
+
+The `kernel.json` file for the `IJulia` kernel should look something like this:
+
+
+```json
+{
+  "display_name": "Julia 1.11.5",
+  "argv": [
+    "/home/user/.julia/juliaup/julia-1.11.5+0.aarch64.apple.darwin14/bin/julia",
+    "-i",
+    "--color=yes",
+    "--project=@.",
+    "/home/user/.julia/packages/IJulia/XF6bn/src/kernel.jl",
+    "{connection_file}"
+  ],
+  "language": "julia",
+  "env": {},
+  "interrupt_mode": "signal"
+}
+```
+
+Note the reference to the `julia` executable in line 4, and the reference to the `kernel.jl` file in line 8. There isn't much magical about kernels. All that happens when Jupyter starts the kernel based on a specific kernel spec is that it runs the process given by `argv`. That is, it runs `julia` with the given command line arguments. It then expects that it can talk to the resulting process with a specific [messaging protocol](https://jupyter-client.readthedocs.io/en/latest/messaging.html#messaging). Here, the code in `kernel.jl` exposes the implementation of that protocol.
+
+
+## Kernel connection failure tips
+
+Fundamentally, if the `IJulia` kernel fails to connect, it might be due to any of the following issues:
```

Comment:
Could we add a tip about checking the notebook logs to see any error messages? For people running Jupyter from `jupyterlab()` or `notebook()` we can also mention `verbose=true`.